File: pinyougou/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbUser;
import com.pinyougou.user.service.UserService;
import com.pinyougou.utils.PhoneFormatCheckUtils;
import com.pinyougou.vo.PageResult;
import com.pinyougou.vo.Result;
import org.apache.commons.codec.digest.DigestUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

import javax.print.attribute.standard.Destination;
import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.regex.PatternSyntaxException;

@RequestMapping("/user")
@RestController
public class UserController {

    @Reference
    private UserService userService;


    @RequestMapping("/findAll")
    public List<TbUser> findAll() {
        return userService.findAll();
    }

    @GetMapping("/findPage")
    public PageResult findPage(@RequestParam(value = "page", defaultValue = "1") Integer page,
                               @RequestParam(value = "rows", defaultValue = "10") Integer rows) {
        return userService.findPage(page, rows);
    }

    @GetMapping("/getUsername")
    public Map<String, Object> getUsername() {
        Map<String, Object> map = new HashMap<>();
        String username = SecurityContextHolder.getContext().getAuthentication().getName();
        map.put("username", username);
        return map;
    }

    @PostMapping("/add")
    public Result add(String smsCode, @RequestBody TbUser user) {
        try {
            if (PhoneFormatCheckUtils.isPhoneLegal(user.getPhone())) {
                //验证验证码是否合法
                if (userService.checkCode(smsCode, user.getPhone())) {
                    //加密
                    user.setPassword(DigestUtils.md5Hex(user.getPassword()));
                    user.setUpdated(new Date());
                    user.setCreated(user.getUpdated());
                    userService.add(user);
                } else {
                    return Result.fail("验证码输入有误");
                }
            } else {
                return Result.fail("手机格式不合法");
            }
            return Result.ok("注册成功");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail("注册失败");
    }

    @GetMapping("/findOne")
    public TbUser findOne(Long id) {
        return userService.findById(id);
    }

    @PostMapping("/update")
    public Result update(@RequestBody TbUser user) {
        try {
            userService.update(user);
            return Result.ok("修改成功");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail("修改失败");
    }

    @GetMapping("/delete")
    public Result delete(Long[] ids) {
        try {
            userService.delete(ids);
            return Result.ok("删除成功");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail("删除失败");
    }

    /**
     * 分页查询列表
     *
     * @param user 查询条件
     * @param page 页号
     * @param rows 每页大小
     * @return
     */
    @PostMapping("/search")
    public PageResult search(@RequestBody TbUser user, @RequestParam(value = "page", defaultValue = "1") Integer page,
                             @RequestParam(value = "rows", defaultValue = "10") Integer rows) {
        return userService.search(page, rows, user);
    }

    /**
     * 发送验证码
     *
     * @param phone
     * @return
     */
    @RequestMapping("/sendSmsCode")
    public Result sendSmsCode(String phone) {
        try {
            if (PhoneFormatCheckUtils.isPhoneLegal(phone)) {
                userService.sendSmsCode(phone);
                return Result.ok("发送验证码成功");
            } else {
                return Result.fail("手机格式错误,发送验证码失败");

            }
        } catch (PatternSyntaxException e) {
            e.printStackTrace();
        }
        return Result.fail("验证码发送失败");
    }
}

File: pinyougou/pinyougou-user-web/src/main/webapp/js/controller/userController.js
app.controller("userController", function ($scope, $controller, userService) {

    $scope.entity = {"username":"", "password":"", "phone":""};
    $scope.smsCode="";
    $scope.register = function () {
        if ($scope.entity.username == "") {
            alert("请输入用户名");
            return;
        }
        if ($scope.entity.username == "anonymousUser") {
            alert("用户名非法");
            return;
        }
        if ($scope.entity.password == "") {
            alert("请输入密码");
            return;
        }
        //判断两次密码是否一致
        if($scope.password != $scope.entity.password){
            alert("两次输入的密码不一致;请重新输入");
            return;
        }

        if ($scope.entity.phone == "") {
            alert("请输入手机号");
            return;
        }
        if ($scope.smsCode == "") {
            alert("请输入验证码");
            return;
        }

        userService.register($scope.entity, $scope.smsCode).success(function (response) {
            $scope.entity = {"username":"", "password":"", "phone":""};
            $scope.smsCode="";
            $scope.password="";
            alert(response.message);
        });
    };

    $scope.entity = {"phone":""};
    $scope.sendSmsCode = function () {
        if($scope.entity.phone == null || $scope.entity.phone=="") {
            alert("请输入手机号");
            return;
        }

        userService.sendSmsCode($scope.entity.phone).success(function (response) {
            alert(response.message);
        });
    };
});